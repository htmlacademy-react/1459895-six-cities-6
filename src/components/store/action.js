export const ActionType = {
  SET_CITY: `main/setCity`,
};

export const ActionCreator = {
  setCity: (city) => ({
    type: ActionType.SET_CITY,
    payload: city,
  }),
};
