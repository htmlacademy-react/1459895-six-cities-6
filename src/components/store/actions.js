export const SET_CITY = `main/setCity`;
export const SET_OPTION = `main/setOption`;
