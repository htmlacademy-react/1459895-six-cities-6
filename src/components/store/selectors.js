import {createSelector} from "reselect";

const getCity = (state) => state.city;
const getOffers = (state) => state.offers;
const getOption = (state) => state.option;
const getReviews = (state) => state.reviews;

export const getActiveOffers = createSelector(
    [getCity, getOffers, getOption],
    (city, offers, option) => {
      const activeOffers = offers.filter((offer) => offer.city.name === city);

      switch (option) {
        case `Price: low to high`:
          return activeOffers.sort((a, b) => a.price - b.price);
        case `Price: high to low`:
          return activeOffers.sort((a, b) => b.price - a.price);
        case `Top rated first`:
          return activeOffers.sort((a, b) => b.rating - a.rating);
        default:
          return activeOffers;
      }
    }
);

export const getActiveReviews = createSelector(
    [getReviews],
    (reviews) => {
      const activeReviews = reviews.sort((a, b) => new Date(b.date) - new Date(a.date));

      return activeReviews.length > 10 ? activeReviews.slice(0, 10) : activeReviews;
    }
);
