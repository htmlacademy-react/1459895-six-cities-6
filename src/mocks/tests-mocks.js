export const TEST_DATA = `test`;

export const cards = {
  start: [
    {
      id: 1,
      isFavorite: false
    },
    {
      id: 2,
      isFavorite: false
    },
    {
      id: 3,
      isFavorite: true
    }
  ],
  end: [
    {
      id: 1,
      isFavorite: false
    },
    {
      id: 2,
      isFavorite: true
    },
    {
      id: 3,
      isFavorite: true
    }
  ],
  change: {
    id: 2,
    isFavorite: true
  }
};

export const favoritesCards = {
  start: [
    {
      id: 1,
      isFavorite: true
    },
    {
      id: 2,
      isFavorite: true
    },
    {
      id: 3,
      isFavorite: true
    }
  ],
  end: [
    {
      id: 2,
      isFavorite: true
    },
    {
      id: 3,
      isFavorite: true
    }
  ],
  remove: {
    id: 1,
    isFavorite: true
  }
};

export const offers = {
  adapted: [
    {
      city: {
        name: `Paris`,
        location: {
          latitude: 48.85661,
          longitude: 2.351499,
          zoom: 13
        }
      },
      images: [
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      ],
      goods: [
        `Laptop friendly workspace`,
        `Washer`,
        `Breakfast`,
        `Air conditioning`
      ],
      host: {
        id: 25,
        name: `Angelina`,
        avatarUrl: `img/avatar-angelina.jpg`,
        isPro: true
      },
      location: {
        latitude: 48.85761,
        longitude: 2.358499,
        zoom: 16
      },
      title: `Tile House`,
      rating: 2,
      type: `house`,
      bedrooms: 2,
      price: 161,
      description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      id: 1,
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      isPremium: true,
      isFavorite: false,
      maxAdults: 9
    },
    {
      city: {
        name: `Amsterdam`,
        location: {
          latitude: 48.85661,
          longitude: 2.351499,
          zoom: 13
        }
      },
      images: [
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      ],
      goods: [
        `Laptop friendly workspace`,
        `Washer`,
        `Breakfast`,
        `Air conditioning`
      ],
      host: {
        id: 25,
        name: `Angelina`,
        avatarUrl: `img/avatar-angelina.jpg`,
        isPro: true
      },
      location: {
        latitude: 48.85761,
        longitude: 2.358499,
        zoom: 16
      },
      title: `Tile House`,
      rating: 4,
      type: `house`,
      bedrooms: 3,
      price: 161,
      description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      id: 2,
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      isPremium: true,
      isFavorite: false,
      maxAdults: 3
    }
  ],
  server: [
    {
      "city": {
        "name": `Paris`,
        "location": {
          "latitude": 48.85661,
          "longitude": 2.351499,
          "zoom": 13
        }
      },
      "images": [
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      ],
      "goods": [
        `Laptop friendly workspace`,
        `Washer`,
        `Breakfast`,
        `Air conditioning`
      ],
      "host": {
        "id": 25,
        "name": `Angelina`,
        "avatar_url": `img/avatar-angelina.jpg`,
        "is_pro": true
      },
      "location": {
        "latitude": 48.85761,
        "longitude": 2.358499,
        "zoom": 16
      },
      "title": `Tile House`,
      "rating": 2,
      "type": `house`,
      "bedrooms": 2,
      "price": 161,
      "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      "id": 1,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      "is_premium": true,
      "is_favorite": false,
      "max_adults": 9
    },
    {
      "city": {
        "name": `Amsterdam`,
        "location": {
          "latitude": 48.85661,
          "longitude": 2.351499,
          "zoom": 13
        }
      },
      "images": [
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
      ],
      "goods": [
        `Laptop friendly workspace`,
        `Washer`,
        `Breakfast`,
        `Air conditioning`
      ],
      "host": {
        "id": 25,
        "name": `Angelina`,
        "avatar_url": `img/avatar-angelina.jpg`,
        "is_pro": true
      },
      "location": {
        "latitude": 48.85761,
        "longitude": 2.358499,
        "zoom": 16
      },
      "title": `Tile House`,
      "rating": 4,
      "type": `house`,
      "bedrooms": 3,
      "price": 161,
      "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      "id": 2,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      "is_premium": true,
      "is_favorite": false,
      "max_adults": 3
    }
  ]
};

export const reviews = {
  server: [
    {
      "user": {
        "id": 11,
        "name": `Jack`,
        "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg`,
        "is_pro": false
      },
      "id": 1,
      "rating": 3,
      "comment": `I stayed here for one night and it was an unpleasant experience.`,
      "date": `2021-02-18T15:34:01.286Z`
    },
    {
      "user": {
        "id": 11,
        "name": `Jack`,
        "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg`,
        "is_pro": false
      },
      "id": 2,
      "rating": 4,
      "comment": `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.`,
      "date": `2021-02-18T15:34:01.286Z`
    }
  ],
  adapted: [
    {
      user: {
        id: 11,
        name: `Jack`,
        avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg`,
        isPro: false
      },
      id: 1,
      rating: 3,
      comment: `I stayed here for one night and it was an unpleasant experience.`,
      date: `2021-02-18T15:34:01.286Z`
    },
    {
      user: {
        id: 11,
        name: `Jack`,
        avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg`,
        isPro: false
      },
      id: 2,
      rating: 4,
      comment: `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.`,
      date: `2021-02-18T15:34:01.286Z`
    }
  ]
};

export const offer = {
  adapted: {
    city: {
      name: `Paris`,
      location: {
        latitude: 48.85661,
        longitude: 2.351499,
        zoom: 13
      }
    },
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`
    ],
    goods: [
      `Laptop friendly workspace`,
      `Washer`,
      `Breakfast`,
      `Air conditioning`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatarUrl: `img/avatar-angelina.jpg`,
      isPro: true
    },
    location: {
      latitude: 48.85761,
      longitude: 2.358499,
      zoom: 16
    },
    title: `Tile House`,
    rating: 2,
    type: `house`,
    bedrooms: 2,
    price: 161,
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    id: 1,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    isPremium: true,
    isFavorite: false,
    maxAdults: 9
  },
  server: {
    "city": {
      "name": `Paris`,
      "location": {
        "latitude": 48.85661,
        "longitude": 2.351499,
        "zoom": 13
      }
    },
    "images": [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    ],
    "goods": [
      `Laptop friendly workspace`,
      `Washer`,
      `Breakfast`,
      `Air conditioning`
    ],
    "host": {
      "id": 25,
      "name": `Angelina`,
      "avatar_url": `img/avatar-angelina.jpg`,
      "is_pro": true
    },
    "location": {
      "latitude": 48.85761,
      "longitude": 2.358499,
      "zoom": 16
    },
    "title": `Tile House`,
    "rating": 2,
    "type": `house`,
    "bedrooms": 2,
    "price": 161,
    "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    "id": 1,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    "is_premium": true,
    "is_favorite": false,
    "max_adults": 9
  },
};

export const authInfo = {
  server: {
    "id": 1,
    "email": `mail@gmail.com`,
    "name": `alex`,
    "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`,
    "is_pro": false
  },
  adapted: {
    id: 1,
    email: `mail@gmail.com`,
    name: `alex`,
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg`,
    isPro: false
  }
};

export const options = [`Popular`, `Price: low to high`, `Price: high to low`, `Top rated first`];
