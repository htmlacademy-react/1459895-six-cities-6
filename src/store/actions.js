export const SET_CITY = `main/setCity`;
export const SET_OPTION = `main/setOption`;
export const SET_AUTHORIZATION = `user/setAuthorization`;
export const SET_OFFERS = `main/setOffers`;
export const SET_NEABY_OFFERS = `proprty/setNearbyOffers`;
export const SET_REVIEWS = `proprty/setReviews`;
export const SET_FAVORITES = `favorite/setFavorites`;
export const REDIRECT_TO_ROUTE = `login/redirectToRoute`;
export const SET_OFFER = `property/setOffer`;
export const SET_IS_LOADED = `property/setIsLoaded`;
export const SET_DISABLED = `property/setDisabled`;
export const SET_IS_ERROR = `property/setIsError`;
export const UPDATE_OFFERS = `main/updateOffers`;
export const UPDATE_FAVORITES = `favorite/updateFavorites`;
export const UPDATE_OFFER = `property/updateOffer`;
export const UPDATE_NEARBY_OFFERS = `property/updateNearbyOffers`;
export const SET_IS_DATA_LOADED = `main/setIsDataLoaded`;
