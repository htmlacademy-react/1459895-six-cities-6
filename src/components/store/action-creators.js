import * as ActionType from "./actions";

export const setCity = (city) => ({type: ActionType.SET_CITY, payload: city});
export const setOption = (option) => ({type: ActionType.SET_OPTION, payload: option});
export const requireAuthorization = (status) => ({type: ActionType.REQUIRED_AUTHORIZATION, payload: status});
export const setOffers = (data) => ({type: ActionType.SET_OFFERS, payload: data});
export const setNearbyOffers = (data) => ({type: ActionType.SET_NEABY_OFFERS, payload: data});
