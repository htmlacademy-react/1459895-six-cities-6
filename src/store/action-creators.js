import * as ActionType from "./actions";
import {createAction} from '@reduxjs/toolkit';

export const setCity = createAction(ActionType.SET_CITY, (city) => {
  return {
    payload: city
  };
});
export const setOption = createAction(ActionType.SET_OPTION, (option) => {
  return {
    payload: option
  };
});
export const setAuthorization = createAction(ActionType.SET_AUTHORIZATION, (data) => {
  return {
    payload: data
  };
});
export const setOffers = createAction(ActionType.SET_OFFERS, (data) => {
  return {
    payload: data
  };
});
export const setNearbyOffers = createAction(ActionType.SET_NEABY_OFFERS, (data) => {
  return {
    payload: data
  };
});
export const setReviews = createAction(ActionType.SET_REVIEWS, (data) => {
  return {
    payload: data
  };
});
export const setFavorite = createAction(ActionType.SET_FAVORITES, (data) => {
  return {
    payload: data
  };
});
export const redirectToRoute = createAction(ActionType.REDIRECT_TO_ROUTE, (url) => {
  return {
    payload: url
  };
});
export const setOffer = createAction(ActionType.SET_OFFER, (data) => {
  return {
    payload: data
  };
});
export const setIsLoaded = createAction(ActionType.SET_IS_LOADED, (isLoaded) => {
  return {
    payload: isLoaded
  };
});
export const setDisabled = createAction(ActionType.SET_DISABLED, (isDisabled) => {
  return {
    payload: isDisabled
  };
});
export const setIsError = createAction(ActionType.SET_IS_ERROR, (isError) => {
  return {
    payload: isError
  };
});

export const updateOffers = createAction(ActionType.UPDATE_OFFERS, (data) => {
  return {
    payload: data
  };
});

export const updateFavorites = createAction(ActionType.UPDATE_FAVORITES, (data) => {
  return {
    payload: data
  };
});

export const updateOffer = createAction(ActionType.UPDATE_OFFER, (data) => {
  return {
    payload: data
  };
});

export const updateNearbyOffers = createAction(ActionType.UPDATE_NEARBY_OFFERS, (data) => {
  return {
    payload: data
  };
});

export const setIsDataLoaded = createAction(ActionType.SET_IS_DATA_LOADED, (isLoad) => {
  return {
    payload: isLoad
  };
});
