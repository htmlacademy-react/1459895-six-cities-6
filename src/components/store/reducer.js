import {ActionType} from "./action";
import {offers} from "../../mocks/offers";

const initialState = {
  city: `Paris`,
  offers: [...offers],
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_CITY:
      return {
        ...state,
        city: action.payload
      };
    default:
      return state;
  }
};

export {reducer};
