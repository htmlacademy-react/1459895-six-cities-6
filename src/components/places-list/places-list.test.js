import React from 'react';
import {render} from '@testing-library/react';
import PlacesList from './places-list';
import {offers, authInfo} from "../../store/tests-mocks";
import * as redux from "react-redux";
import configureStore from "redux-mock-store";
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';

const mockStore = configureStore({});
describe(`Test PlacesList component`, () => {
  it(`PlacesList component should render correctly`, () => {
    const type = `CITIES`;
    const store = mockStore({
      USER: {authInfo: null}
    });
    const history = createMemoryHistory();
    const {container} = render(
        <redux.Provider store={store}>
          <Router history={history}>
            <PlacesList
              type={type}
              offers={offers.adapted}
              onChangeActiveCard={jest.fn()}
              onScrollToTop={jest.fn()}
              onFavoriteClick={jest.fn()}
            />
          </Router>
        </redux.Provider>
    );
    expect(container).toMatchSnapshot();
  });
  it(`PlacesList component should render correctly when user athorized`, () => {
    const type = `CITIES`;
    const store = mockStore({
      USER: {authInfo: authInfo.adapted}
    });
    const history = createMemoryHistory();
    const {container} = render(
        <redux.Provider store={store}>
          <Router history={history}>
            <PlacesList
              type={type}
              offers={offers.adapted}
              onChangeActiveCard={jest.fn()}
              onScrollToTop={jest.fn()}
              onFavoriteClick={jest.fn()}
            />
          </Router>
        </redux.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
