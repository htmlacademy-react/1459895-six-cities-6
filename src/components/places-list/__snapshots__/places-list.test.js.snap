// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PlacesList component PlacesList component should render correctly 1`] = `
<div>
  <div
    class="cities__places-list places__list tabs__content"
  >
    <article
      class="cities__place-card place-card"
    >
      <div
        class="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        class="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/1"
        >
          <img
            alt="Place image"
            class="place-card__image"
            height="200"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        class=" place-card__info"
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
              161
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button button false"
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                href="#icon-bookmark"
                xlink="true"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span
              style="width: 40%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/1"
          >
            Tile House
          </a>
        </h2>
        <p
          class="place-card__type"
        >
          house
        </p>
      </div>
    </article>
    <article
      class="cities__place-card place-card"
    >
      <div
        class="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        class="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/2"
        >
          <img
            alt="Place image"
            class="place-card__image"
            height="200"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        class=" place-card__info"
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
              161
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button button false"
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                href="#icon-bookmark"
                xlink="true"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span
              style="width: 80%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/2"
          >
            Tile House
          </a>
        </h2>
        <p
          class="place-card__type"
        >
          house
        </p>
      </div>
    </article>
  </div>
</div>
`;

exports[`Test PlacesList component PlacesList component should render correctly when user athorized 1`] = `
<div>
  <div
    class="cities__places-list places__list tabs__content"
  >
    <article
      class="cities__place-card place-card"
    >
      <div
        class="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        class="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/1"
        >
          <img
            alt="Place image"
            class="place-card__image"
            height="200"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        class=" place-card__info"
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
              161
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button button false"
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                href="#icon-bookmark"
                xlink="true"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span
              style="width: 40%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/1"
          >
            Tile House
          </a>
        </h2>
        <p
          class="place-card__type"
        >
          house
        </p>
      </div>
    </article>
    <article
      class="cities__place-card place-card"
    >
      <div
        class="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        class="cities__image-wrapper place-card__image-wrapper"
      >
        <a
          href="/offer/2"
        >
          <img
            alt="Place image"
            class="place-card__image"
            height="200"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        class=" place-card__info"
      >
        <div
          class="place-card__price-wrapper"
        >
          <div
            class="place-card__price"
          >
            <b
              class="place-card__price-value"
            >
              €
              161
            </b>
            <span
              class="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            class="place-card__bookmark-button button false"
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                href="#icon-bookmark"
                xlink="true"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          class="place-card__rating rating"
        >
          <div
            class="place-card__stars rating__stars"
          >
            <span
              style="width: 80%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          class="place-card__name"
        >
          <a
            href="/offer/2"
          >
            Tile House
          </a>
        </h2>
        <p
          class="place-card__type"
        >
          house
        </p>
      </div>
    </article>
  </div>
</div>
`;
